app/src/main/res/xml/appearance_settings.xml

```diff
@@ -34,4 +34,10 @@
         android:title="@string/caption_setting_title"
         app:iconSpaceReserved="false" />
 
+    <SwitchPreferenceCompat
```

Comment:
This switch should be hidden on API <21, since the lockscreen thumbnail was not a feature until Android Lollipop

Reply:
Is there a preference for doing this with xml only or programmatically hiding it.

Reply:
You have to do this programmatically by getting the preference object in the settings fragment for appearance and hiding it

Reply:
XML only is also possible, hence my question
You have values/bool.xml with a value being false and values-v22/bool.xml with the same value being true. In the preference xml you then add `app:isPreferenceVisible="@bool/{key}"`
Same could be done for `app:visibility`, with the values being strings for gone, invisible and visible.

Reply:
This looks like obfuscation: even though it works, it's more of a workaround than a real solution. I'd use normal code like it's done in all other `PreferenceFragment`s